packages/new-block-stream/README.md

```diff
@@ -4,6 +4,87 @@ A tool for creating a stream of new block data.
 
 ## Usage
 
+### Getting started
+
+Install via npm:
+
+```js
+npm install @drizzle-utils/new-block-stream
+```
+
+Usage example (taken from `@drizzle-utils/new-block-stream/example.js`)
+
 ```js
-// TODO
+/* eslint-disable no-console */
+const Web3 = require("web3");
+const createNewBlock$ = require("@drizzle-utils/new-block-stream");
+
+const main = async () => {
```

Comment:
We don't need to put this in a `main` function, it's not a runnable example, it's just a usage example. Something like this would be sufficient:

```js
const Web3 = require("web3");
const createNewBlock$ = require("@drizzle-utils/new-block-stream");

const web3 = new Web3("ws://127.0.0.1:9545");
const { observable, subscription } = createNewBlock$({
   web3,
   pollingInterval: 200, // only used if non-WebsocketProvider
 });

 // log out new blocks (might skip blocks if polling)
observable.subscribe(block => console.log(block));
```

Reply:
Hmm I kinda like the `setTimeout` part, as it shows the full lifecycle of the stream with the `unsubscribe`. It's more obvious that the user will eventually, need to unsub at some point in time (e.g. component unmounting). Thoughts?

Reply:
Technically unsubscribing isn't a "natural" lifecycle of using RxJS observables. In fact, ideally you'd want observables that complete on their own. See this article: https://medium.com/@benlesh/rxjs-dont-unsubscribe-6753ed4fda87

Also, if the stream is something that is supposed to run for the entirety of the webapp, there's no real need to unsubscribe.

That being said, our API is actually really confusing, because people are going to need to understand the difference between calling `.unsubscribe` on the stream (subscribed observable) vs calling it on the `subscription` object that is returned. I think we might need a better API for that.

That being said, I don't like the `setTimeout` approach because it adds another layer of indentation/indirection. I'd prefer something like this:

```js
const Web3 = require("web3");
const createNewBlock$ = require("@drizzle-utils/new-block-stream");

const web3 = new Web3("ws://127.0.0.1:9545");
const { observable, subscription } = createNewBlock$({
   web3,
   pollingInterval: 200, // only used if non-WebsocketProvider
 });

 // log out new blocks (might skip blocks if polling)
const newBlock$ = observable.subscribe(block => console.log(block));

/* after some time, you want to stop listening to new blocks */
newBlock$.unsubscribe()     // unsubscribe from the RxJS stream
subscription.unsubscribe()  // unsubscribe from the underlying block listeners
```

In the future, maybe we can return a `cleanUp` function that will complete the observable as well, so there's no need to call `unsubscribe` on two things. All you'd have to do is call "cleanUp()" and you're done.

Reply:
Sounds good @adrianmcli